flows/runs/run.go

```diff
@@ -197,33 +196,29 @@ func (r *flowRun) ExpiresOn() *time.Time       { return r.expiresOn }
 func (r *flowRun) TimesOutOn() *time.Time      { return r.timesOutOn }
 func (r *flowRun) ExitedOn() *time.Time        { return r.exitedOn }
 
-func (r *flowRun) updateTranslations() {
-	if r.flowTranslations != nil {
-		r.translations = r.flowTranslations.GetTranslations(r.language)
-	} else {
-		r.translations = nil
-	}
-}
-func (r *flowRun) SetFlowTranslations(ft flows.FlowTranslations) {
-	r.flowTranslations = ft
-	r.updateTranslations()
-}
-func (r *flowRun) SetLanguage(lang utils.Language) {
-	r.language = lang
-	r.updateTranslations()
+func (r *flowRun) SetLanguages(langs utils.LanguageList) {
+	r.languages = append(langs, r.Flow().Language())
 }
-func (r *flowRun) GetText(uuid flows.UUID, key string, backdown string) string {
-	if r.translations == nil {
-		return backdown
-	}
-	return r.translations.GetText(uuid, key, backdown)
+func (r *flowRun) GetText(uuid flows.UUID, key string, native string) string {
+	textArray := r.GetTextArray(uuid, key, []string{native})
+	return textArray[0]
 }
 
-func (r *flowRun) GetTranslations(uuid flows.UUID, key string, backdown []string) []string {
-	if r.translations == nil {
-		return backdown
+func (r *flowRun) GetTextArray(uuid flows.UUID, key string, native []string) []string {
+	for _, lang := range r.languages {
+		if lang == r.Flow().Language() {
+			return native
```

Comment:
I've renamed `backdown` to `native` because the native text in the flow will often be the preferred language - i.e. not a backdown.